### PR TITLE
VZ-9701 module metadata and charts

### PR DESCRIPTION
**Pull request by jmaron99**
- added index.yaml and charts to docker image
- updated go.mod is most likely due to addition of helm lib (for reading/processing index.yaml file)
- code to generate module-metadata config map

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -345,6 +345,16 @@ docker-pull-prerequisites:
 	docker pull ghcr.io/oracle/oraclelinux:8
 	docker pull ghcr.io/oracle/oraclelinux:8-slim
 
+.PHONY: generate-modules-repo-artifacts
+generate-modules-repo-artifacts:
+	git clone https://github.com/verrazzano/verrazzano-modules.git
```
Comment (mgianatagh) [inline Makefile:350]: Should there be an option to checkout a specific branch?  And the default behavior to be checkout the tip of the default branch defined for the repo?

File: util/ocne/testdata/index.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+apiVersion: v1
+entries:
+  oci-ccm:
+    - apiVersion: v1
+      appVersion: 1.25.0
+      created: "2023-05-18T13:11:16.630779-04:00"
+      description: Oracle Cloud Infrastructure (OCI) cloud controller manager
+      digest: 303db242eb62fe260a3333902706d7960db4770f24d96edcb8929573beee2f9b
+      kubeVersion: ^1.23.0-0
+      maintainers:
```
Comment (mgianatagh) [inline util/ocne/testdata/index.yaml:10]: Does the maintainer info need to be present?

**Review by mgianatagh — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (jmaron99) [reply Makefile:350]: I supposed I could add that logic but I'm not sure about its immediate need.  I could add a comment showing how to do it (add a git checkout :) )

File: util/ocne/testdata/index.yaml (hunk 2 above)
Comment (jmaron99) [reply util/ocne/testdata/index.yaml:10]: It's gleamed by the helm command generating the file from the Chart.yaml which is a publicly available file in the origin repository.  I could remove it from the the test file if necessary but, as I said, it is publicly available information.

File: Makefile (hunk 1 above)
Comment (mgianatagh) [reply Makefile:350]: We can wait until the need arises.

File: util/ocne/testdata/index.yaml (hunk 2 above)
Comment (mgianatagh) [reply util/ocne/testdata/index.yaml:10]: ok, seems ok as is then since it is publicly available information.

**Review by theabmitra — APPROVED** [review]

**Pull request merged**